### PR TITLE
fix mdbook address issue #202

### PR DESCRIPTION
**Pull request by gpwclark**
main change is when/how the usage function is loaded. Other than that just made some changes so that core/test were available to all the code loaded in `slosh` code blocks.

### CODE REVIEW

**Review by StevenLove — APPROVED** [review]
looks good!

File: doc/src/syntax-and-macros.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Expressions are enclosed in parentheses and Lisp uses prefix notation to parse expressions.
+A parenthesized expression is a list of forms: `(list 1 2 3 4)` where the first
+form, the head, should generally be a function or a macro and dictates what the expression will do,
+and the subsequent children are arguments to the head.
+
+## Comments
+
+Single-line comments start with a ; character and continue till the end of the line.
+Multi-line comments start with a #| and end with a |#
```
Comment (sstanfield) [inline doc/src/syntax-and-macros.md:29]: Also #; will consume and discard the next 'thing', usually an sexp- mostly for debugging.  Also could mention #%...%# for docstrings but those are special and not general comments.

File: doc/src/section-docs/hashmap.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+(set! (get m "three") "changed3")
+(assert-equal "changed3" (get m "three"))
+```
+Clearing all keys is done with the `clear!` function.
+```slosh
+(def m {:one "one", :two "two", "three" 3})
+(clear! m)
+(assert-equal (make-hash) m)
+```
+** Hashmap keys that are not keyword types (not prefixed by a colon) can not be
```
Comment (sstanfield) [inline doc/src/section-docs/hashmap.md:30]: This is a bug, you should (and one point) could do this.  At least I think you could.  At any rate I think you should be able to do this...

File: doc/src/section-docs/hashmap.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+`set!`, and `clear!`.
+
+For retrieving items from a hashmap prefer the reader syntax (<object>.<keyword>) or use the `get`
+function.
+```slosh
+(def m {:one "one", :two "two", "three" 3})
+(assert-equal "one" m.:one)
+(assert-equal "two" m.:two)
+(assert-equal 3 (get m "three"))
+```
```
Comment (sstanfield) [inline doc/src/section-docs/hashmap.md:12]: Might mention the ~symbol support here.  For instance '(def one :one) (pro m.~one) should work to allow the use of a variable after the dot (need this for looping with int keys for instance).

**Pull request merged**